src/herder/TransactionQueue.cpp

```diff
@@ -1061,73 +1083,59 @@ TransactionQueue::broadcastSome()
     }
 
     std::vector<TransactionFrameBasePtr> banningTxs;
-    while (opsToFlood != 0 && !iters.empty())
-    {
-        auto curTracker = iters.top();
-        iters.pop();
+    auto visitor = [this, &totalOpsToFlood,
+                    &banningTxs](TxStack const& txStack) {
+        auto const& curTracker = static_cast<TxQueueTracker const&>(txStack);
         // look at the next candidate transaction for that account
-        auto& cur = curTracker.mCur;
-        auto& tx = cur->mTx;
+        auto& cur = curTracker.getCurrentTimestampedTx();
+        auto tx = curTracker.getTopTx();
         // by construction, cur points to non broadcasted transactions
-        releaseAssert(!cur->mBroadcasted);
-        if (opsToFlood < tx->getNumOperations())
-        {
-            // as we can't flood this transaction we have to wait to
-            // accumulate more flooding credit
-            break;
-        }
-        auto bStatus = broadcastTx(*curTracker.mAccountState, *cur);
+        releaseAssert(!cur.mBroadcasted);
+        auto bStatus = broadcastTx(curTracker.getAccountState(), cur);
         if (bStatus == BroadcastStatus::BROADCAST_STATUS_SUCCESS)
         {
-            auto ops = tx->getNumOperations();
-            opsToFlood -= ops;
-            totalOpsToFlood -= ops;
+            totalOpsToFlood -= tx->getNumOperations();
+            return SurgePricingPriorityQueue::VisitTxStackResult::TX_PROCESSED;
         }
-        // when skipping, we ban the transaction and skip the remainder of the
-        // queue
-        if (bStatus == BroadcastStatus::BROADCAST_STATUS_SKIPPED)
+        else if (bStatus == BroadcastStatus::BROADCAST_STATUS_SKIPPED)
         {
+            // When skipping, we ban the transaction and skip the remainder of
+            // the stack.
             banningTxs.emplace_back(tx);
+            return SurgePricingPriorityQueue::VisitTxStackResult::
+                TX_STACK_SKIPPED;
         }
         else
         {
-            cur++;
-            // if we're not done with this account, add the tracker back
-            if (curTracker.skipToFirstNotBroadcasted())
-            {
-                iters.push(curTracker);
-            }
+            // Already broadcasted; don't invalidate the stack but also don't
+            // count transaction as processed.
+            return SurgePricingPriorityQueue::VisitTxStackResult::TX_SKIPPED;
         }
-    }
+    };
+    SurgePricingPriorityQueue::visitTopTxs(trackersToBroadcast, opsToFlood,
+                                           mBroadcastSeed, visitor,
+                                           mBroadcastOpCarryover);
     ban(banningTxs);
     // carry over remainder, up to MAX_OPS_PER_TX ops
     // reason is that if we add 1 next round, we can flood a "worst case fee
     // bump" tx
-    mBroadcastOpCarryover = std::min<size_t>(opsToFlood, MAX_OPS_PER_TX + 1);
+    mBroadcastOpCarryover = std::min(mBroadcastOpCarryover, MAX_OPS_PER_TX + 1);
```

Comment:
changes in this file may conflict with the pull mode PR (I just checked and changes look to be in different places, so it will depend on how good auto-merge is)

Reply:
The queue has been merged without conflicts.